towerlib/entities/host.py

```diff
@@ -318,7 +317,7 @@ def associate_with_groups(self, groups):
         """
         if not isinstance(groups, (list, tuple)):
             groups = [groups]
-        inventory_groups = [group for group in self.inventory.groups]
+        inventory_groups = self.inventory.groups
```

Comment:
self.inventory.groups is a generator not a list. Did not check where inventory_groups are used but i have a feeling that making the self.inventory.groups into a list served a purpose, please make sure that it does not before we remove this.

Reply:
Just checked, the way the code is written we don't need the intermediate ```inventory_groups``` variable, we  just refer to ```self.inventory.groups``` . I think this is a remnant of when the code was not optimized.

Reply:
Changed to self.inventory.groups instead of inventory_groups variable.